packages/renderer/src/lib/appearance/appearance-util.ts

```diff
@@ -64,10 +48,10 @@ export class AppearanceUtil {
       return icon;
     }
 
-    const isDark: boolean = await this.isDarkMode();
-    if (isDark && icon.dark) {
+    const isDarkTheme = get(isDark);
```

Comment:
You are exploiting that isDark is a store.
I suppose that we need to transform the `getImage` function with a function to which the caller can subscribe to get new values of the image when `isDark` changes (by subscribing to `isDark` here, and re-computing the image and re-emitting its value)

Reply:
(same as Florent's comment)